src/GafferImage/Merge.cpp

```diff
@@ -203,6 +237,33 @@ struct MergeFunctor
 
 };
 
+struct MergeDataWindowFunctor
+{
+	typedef void ReturnType;
+
+	template<class Op>
+	ReturnType operator()( Box2i &existingWindow, const Box2i &newWindow )
+	{
+		if( Op::foregroundOnly == Black && Op::backgroundOnly == Black )
+		{
+			existingWindow = BufferAlgo::intersection( existingWindow, newWindow );
+		}
+		else if( Op::foregroundOnly == Black )
+		{
+			// Leave the existingWindow
+		}
+		else if( Op::backgroundOnly == Black )
+		{
+			existingWindow = newWindow;
+		}
+		else
+		{
+			existingWindow.extendBy( newWindow );
+		}
+	}
+
+};
```

Comment:
You mentioned not being totally happy with the naming of things at one point. If I've understood this function correctly, I think it might be easier to follow if it returned the result, and if `existingWindow` was called `backgroundWindow` and `newWindow` was called `foregroundWindow`.